### PR TITLE
Revert "feat(table): allow conditionally showing action items"

### PR DESCRIPTION
**Pull request by brc-dd**
Reverts globalbrain/sefirot#374

do something like instead:

```ts
    actions: {
      cell: (_, record) => { /* conditional logic here */ }
    }
```

### CODE REVIEW

File: docs/components/table.md [hunk 1]
```diff
@@ -510,10 +510,7 @@ interface TableCellActions {
   labelMode?: ColorModes
 
   // Callback function when the button is clicked.
-  onClick(record: any): void
-
-  // Whether to show the button for the record.
-  show?(record: any): boolean
+  onClick(value: any, record: any): void
```
Comment (brc-dd) [inline docs/components/table.md:513]:
```suggestion
  onClick(record: any): void
```

File: lib/components/STableCell.vue [hunk 2]
```diff
@@ -102,6 +102,7 @@ const computedCell = computed<TableCell | undefined>(() =>
     />
     <STableCellActions
       v-else-if="computedCell.type === 'actions'"
+      :value="value"
```
Comment (brc-dd) [inline lib/components/STableCell.vue:105]:
```suggestion
```

File: lib/components/STableCellActions.vue [hunk 3]
```diff
@@ -3,26 +3,26 @@ import { type TableCellAction } from '../composables/Table'
 import SButton from './SButton.vue'
 
 defineProps<{
+  value: any
```
Comment (brc-dd) [inline lib/components/STableCellActions.vue:6]:
```suggestion
```

File: lib/components/STableCellActions.vue [hunk 4]
```diff
@@ -3,26 +3,26 @@ import { type TableCellAction } from '../composables/Table'
...
+      v-for="(action, i) in actions"
+      :key="i"
+      size="mini"
+      type="text"
+      :mode="action.mode ?? 'mute'"
+      :icon="action.icon"
+      :icon-mode="action.iconMode"
+      :label="action.label"
+      :label-mode="action.labelMode"
+      @click="action.onClick(value, record)"
```
Comment (brc-dd) [inline lib/components/STableCellActions.vue:24]:
```suggestion
      @click="action.onClick(record)"
```

File: lib/composables/Table.ts [hunk 5]
```diff
@@ -189,8 +189,7 @@ export interface TableCellAction {
   iconMode?: Mode
   label?: string
   labelMode?: Mode
-  onClick(record: any): void
-  show?(record: any): boolean
+  onClick(value: any, record: any): void
```
Comment (brc-dd) [inline lib/composables/Table.ts:192]:
```suggestion
  onClick(record: any): void
```

**Comment by brc-dd** [comment]
I'm not sure about this either. With the `cell.show` syntax I have an idea to fix #377 (-- we can calculate the width of labels and icons and add them + padding). Having cell functions would mess things a bit.

**Comment by kiaking** [comment]
@brc-dd 
Yeah true. If we remove the actions from the array, it would be hard to calculate the desired width of the actions cell automatically 🤔 

It might be hard for everybody to follow the guideline on when to use `cell()` function and when to use `show` property, but I think `cell()` function best works when we know certain actions are not available all the time.

For example, if delete icon should only be visible to Admin, then guest users should only ever see 1 icon. In this case, we should just use `cell()` and add only 1 item in the array.

On the other hand, if the action is conditional based on each record, then we should be using `show` I guess.

So, I think having `show` at the moment make sense. What do you think 🤔

**Pull request closed without merging**